src/csprojWriter.ts

```diff
@@ -0,0 +1,227 @@
+import * as path from 'path';
+import { promises as fs } from 'fs';
+
+import * as findupglob from 'find-up-glob';
+import { Builder, Parser } from 'xml2js';
+
+export enum BuildActions {
+    Folder = 'Folder',
+    Compile = 'Compile',
+    Content = 'Content',
+    EmbeddedResource = 'EmbeddedResource',
+    PRIResource = 'PRIResource',
+    Page = 'Page',
+    None = 'None',
+}
+
+export class CsprojWriter {
+    public async getProjFilePath(filePath: string): Promise<string | undefined> {
+        const projItems: string[] = await findupglob('*.projitems', { cwd: path.dirname(filePath) });
+        const csProj: string[] = await findupglob('*.csproj', { cwd: path.dirname(filePath) });
+
+        if (projItems !== null && projItems.length >= 1) return projItems[0];
+        else if (csProj !== null && csProj.length >= 1) return csProj[0];
```

Comment:
Else is not necessary.

Reply:
Thank you for finding these issues in coding-style. I think the style is more consistent now